core/builder/src/main/java/io/quarkus/builder/Json.java

```diff
@@ -26,6 +26,10 @@ public final class Json {
     private static final int CONTROL_CHAR_START = 0;
     private static final int CONTROL_CHAR_END = 0x1f;
     private static final char CHAR_QUOTATION_MARK = '"';
+    private static final String DEFAULT_INDENT = "  ";
+    private static final String EMPTY_STRING = "";
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final String ENTRY_SEPARATOR_NEW_LINE = "," + NEW_LINE;
```

Comment:
See @vsevel's https://github.com/quarkusio/quarkus/pull/36826#issuecomment-1839160473 comment for motivation for displaying the generated configuration.

I can see the benefit of displaying the configuration as part of the build process, as well as emitting actual files. I agree that it's a pain having to add the pretty printing logic. The best alternative I can think for anyone that wants to display the generated configuration file without the pretty printing logic would to configure CI such that:

1. Runs unit tests with `native-with-agent` (e.g. `mvn test`)
2. Install `jq` or similar
3. Invoke `jq` to pretty print the json files.
4. Run further build steps without re-running unit tests.

Steps 2/3 could be moved to a latter stage depending on the complexity of doing step 4. @vsevel how does this sound?

Note about increasing the configuration size. The configuration files produced by the native image agent are generated pretty printed. What I have added is pretty printing for the json files that are post-transformed, which is the resource configuration json file. I did that because the JSON that comes out of the builder JSON printed is not formatted.

Reply:
Oh, I forgot about @vsevel's request. I leave this up to you two. I am OK with pretty printing, I was just concerned about the increased complexity.